### PR TITLE
Add CI job with GCC 14

### PR DESCRIPTION
**Pull request by SiarheiFedartsou**
## Benchmark Results
| Benchmark | Base | PR |
|-----------|------|----|
| alias | aliased u32: 1136.92<br/>plain u32: 1144.72<br/>aliased double: 1170.81<br/>plain double: 1177.3 | aliased u32: 1148.08<br/>plain u32: 1144.8<br/>aliased double: 1179.63<br/>plain double: 1176.99 |
| json-render | String: 6.78285ms<br/>Stringstream: 10.7539ms<br/>Vector: 6.63403ms | String: 6.77858ms<br/>Stringstream: 10.6641ms<br/>Vector: 6.60302ms |
| match_ch | Default radius: <br/>4.49168ms/req at 82 coordinate<br/>0.0547766ms/coordinate<br/>Radius 5m: <br/>4.48498ms/req at 82 coordinate<br/>0.0546949ms/coordinate<br/>Radius 10m: <br/>15.2949ms/req at 82 coordinate<br/>0.186524ms/coordinate<br/>Radius 15m: <br/>37.2586ms/req at 82 coordinate<br/>0.454373ms/coordinate<br/>Radius 30m: <br/>318.68ms/req at 82 coordinate<br/>3.88635ms/coordinate | Default radius: <br/>4.52378ms/req at 82 coordinate<br/>0.0551681ms/coordinate<br/>Radius 5m: <br/>4.49523ms/req at 82 coordinate<br/>0.0548198ms/coordinate<br/>Radius 10m: <br/>15.2356ms/req at 82 coordinate<br/>0.1858ms/coordinate<br/>Radius 15m: <br/>37.2456ms/req at 82 coordinate<br/>0.454215ms/coordinate<br/>Radius 30m: <br/>318.86ms/req at 82 coordinate<br/>3.88854ms/coordinate |
| match_mld | Default radius: <br/>2.89378ms/req at 82 coordinate<br/>0.0352899ms/coordinate<br/>Radius 5m: <br/>2.93111ms/req at 82 coordinate<br/>0.0357453ms/coordinate<br/>Radius 10m: <br/>10.6146ms/req at 82 coordinate<br/>0.129446ms/coordinate<br/>Radius 15m: <br/>26.5207ms/req at 82 coordinate<br/>0.323423ms/coordinate<br/>Radius 30m: <br/>310.109ms/req at 82 coordinate<br/>3.78182ms/coordinate | Default radius: <br/>2.89575ms/req at 82 coordinate<br/>0.035314ms/coordinate<br/>Radius 5m: <br/>2.87896ms/req at 82 coordinate<br/>0.0351092ms/coordinate<br/>Radius 10m: <br/>10.5363ms/req at 82 coordinate<br/>0.128492ms/coordinate<br/>Radius 15m: <br/>27.0866ms/req at 82 coordinate<br/>0.330324ms/coordinate<br/>Radius 30m: <br/>311.635ms/req at 82 coordinate<br/>3.80043ms/coordinate |
| packedvector | random write:<br/>std::vector 11233.1 ms<br/>util::packed_vector 74128.1 ms<br/>slowdown: 6.59908<br/>random read:<br/>std::vector 11070.5 ms<br/>util::packed_vector 31024.9 ms<br/>slowdown: 2.80247 | random write:<br/>std::vector 11234 ms<br/>util::packed_vector 78482.4 ms<br/>slowdown: 6.98613<br/>random read:<br/>std::vector 11063 ms<br/>util::packed_vector 33914 ms<br/>slowdown: 3.06553 |
| route_ch | 1000 routes, 3 coordinates, no alternatives, overview=full, steps=true<br/>601.194ms<br/>0.601194ms/req<br/>1000 routes, 2 coordinates, no alternatives, overview=full, steps=true<br/>383.515ms<br/>0.383515ms/req<br/>1000 routes, 2 coordinates, 3 alternatives, overview=full, steps=true<br/>768.634ms<br/>0.768634ms/req<br/>1000 routes, 3 coordinates, no alternatives, overview=false, steps=false<br/>162.246ms<br/>0.162246ms/req<br/>1000 routes, 2 coordinates, no alternatives, overview=false, steps=false<br/>103.308ms<br/>0.103308ms/req<br/>1000 routes, 2 coordinates, 3 alternatives, overview=false, steps=false<br/>146.042ms<br/>0.146042ms/req<br/>1000 routes, 3 coordinates, no alternatives, overview=false, steps=false, radius=750<br/>160.782ms<br/>0.160782ms/req<br/>1000 routes, 2 coordinates, no alternatives, overview=false, steps=false, radius=750<br/>103.191ms<br/>0.103191ms/req<br/>1000 routes, 2 coordinates, 3 alternatives, overview=false, steps=false, radius=750<br/>144.365ms<br/>0.144365ms/req | 1000 routes, 3 coordinates, no alternatives, overview=full, steps=true<br/>592.712ms<br/>0.592712ms/req<br/>1000 routes, 2 coordinates, no alternatives, overview=full, steps=true<br/>374.102ms<br/>0.374102ms/req<br/>1000 routes, 2 coordinates, 3 alternatives, overview=full, steps=true<br/>755.684ms<br/>0.755684ms/req<br/>1000 routes, 3 coordinates, no alternatives, overview=false, steps=false<br/>153.186ms<br/>0.153186ms/req<br/>1000 routes, 2 coordinates, no alternatives, overview=false, steps=false<br/>98.3703ms<br/>0.0983703ms/req<br/>1000 routes, 2 coordinates, 3 alternatives, overview=false, steps=false<br/>134.762ms<br/>0.134762ms/req<br/>1000 routes, 3 coordinates, no alternatives, overview=false, steps=false, radius=750<br/>152.69ms<br/>0.15269ms/req<br/>1000 routes, 2 coordinates, no alternatives, overview=false, steps=false, radius=750<br/>98.3767ms<br/>0.0983767ms/req<br/>1000 routes, 2 coordinates, 3 alternatives, overview=false, steps=false, radius=750<br/>134.115ms<br/>0.134115ms/req |
| route_mld | 1000 routes, 3 coordinates, no alternatives, overview=full, steps=true<br/>729.377ms<br/>0.729377ms/req<br/>1000 routes, 2 coordinates, no alternatives, overview=full, steps=true<br/>472.8ms<br/>0.4728ms/req<br/>1000 routes, 2 coordinates, 3 alternatives, overview=full, steps=true<br/>959ms<br/>0.959ms/req<br/>1000 routes, 3 coordinates, no alternatives, overview=false, steps=false<br/>293.97ms<br/>0.29397ms/req<br/>1000 routes, 2 coordinates, no alternatives, overview=false, steps=false<br/>173.252ms<br/>0.173252ms/req<br/>1000 routes, 2 coordinates, 3 alternatives, overview=false, steps=false<br/>317.196ms<br/>0.317196ms/req<br/>1000 routes, 3 coordinates, no alternatives, overview=false, steps=false, radius=750<br/>294.635ms<br/>0.294635ms/req<br/>1000 routes, 2 coordinates, no alternatives, overview=false, steps=false, radius=750<br/>173.579ms<br/>0.173579ms/req<br/>1000 routes, 2 coordinates, 3 alternatives, overview=false, steps=false, radius=750<br/>316.162ms<br/>0.316162ms/req | 1000 routes, 3 coordinates, no alternatives, overview=full, steps=true<br/>723.105ms<br/>0.723105ms/req<br/>1000 routes, 2 coordinates, no alternatives, overview=full, steps=true<br/>465.076ms<br/>0.465076ms/req<br/>1000 routes, 2 coordinates, 3 alternatives, overview=full, steps=true<br/>942.83ms<br/>0.94283ms/req<br/>1000 routes, 3 coordinates, no alternatives, overview=false, steps=false<br/>280.501ms<br/>0.280501ms/req<br/>1000 routes, 2 coordinates, no alternatives, overview=false, steps=false<br/>163.215ms<br/>0.163215ms/req<br/>1000 routes, 2 coordinates, 3 alternatives, overview=false, steps=false<br/>302.271ms<br/>0.302271ms/req<br/>1000 routes, 3 coordinates, no alternatives, overview=false, steps=false, radius=750<br/>278.507ms<br/>0.278507ms/req<br/>1000 routes, 2 coordinates, no alternatives, overview=false, steps=false, radius=750<br/>163.844ms<br/>0.163844ms/req<br/>1000 routes, 2 coordinates, 3 alternatives, overview=false, steps=false, radius=750<br/>307.214ms<br/>0.307214ms/req |
| rtree | 1 result:<br/>207.601ms ->  0.0207601 ms/query<br/>10 results:<br/>242.511ms ->  0.0242511 ms/query | 1 result:<br/>206.572ms ->  0.0206572 ms/query<br/>10 results:<br/>242.554ms ->  0.0242554 ms/query |

### CODE REVIEW

File: scripts/update_dependencies.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ SOL_PATH="ThePhD/sol2"
 SOL_TAG=v2.17.5
 
 RAPIDJSON_PATH="Tencent/rapidjson"
-RAPIDJSON_TAG=v1.1.0
+RAPIDJSON_TAG=f9d53419e912910fd8fa57d5705fa41425428c35
```
Comment (SiarheiFedartsou) [inline scripts/update_dependencies.sh:22]: This change is extracted to separate PR https://github.com/Project-OSRM/osrm-backend/pull/6906

**Review by DennisOSRM — APPROVED** [review]
